obspy/io/y/core.py

```diff
@@ -256,10 +307,11 @@ def _read_y(filename, headonly=False, **kwargs):  # @UnusedVariable
                 # UCHAR Key[16]
                 #   Unique key that identifies this record in the database.
                 data = fh.read(next_tag)
-                parts = unpack(endian + b'd16s', data[8:])
+                parts = _unpack_with_asciiz_and_decode(
+                    endian + b'd16s', data[8:])
```

Comment:
Unfortunately the [manual](https://github.com/obspy/obspy/blob/master/obspy/io/y/docs/NmxDataFormats_RefGuide_14602R4.pdf) is not very stringent. For many `char` fields it doesn't specify whether it's `BLANKPAD` or `ASCIIZ`. I decided to only touch things that have a clear statement in the manual whether it's  `BLANKPAD` or `ASCIIZ` (I also didn't want to touch more than neccessary since I think that header fields are more or less untested).